corehq/apps/data_dictionary/views.py

```diff
@@ -60,6 +62,18 @@
 )
 
 
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def count_cases_for_case_type(request, domain, case_type_name):
+    case_property = request.GET.get('case_property', '')
+    if case_property:
+        query = CaseSearchES().domain(domain).case_type(case_type_name)
+        case_count = query.NOT(case_property_missing(case_property)).count()
+    else:
+        case_count = CaseSearchES().domain(domain).case_type(case_type_name).count()
```

Comment:
Nit: Since both queries are largely the same, this could be simplified a little
```suggestion
    query = CaseSearchES().domain(domain).case_type(case_type_name)
    if case_property:
        query = query.NOT(case_property_missing(case_property))
    case_count = query.count()
```